test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -2008,6 +2024,101 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Waiting on pod to be Ready")
+				running, err := pod.WaitOnSuccesses(deploymentName, "default", 4, sleepBetweenRetriesWhenWaitingForPodReady, cfg.Timeout)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(running).To(Equal(true))
+
+				By("Exposing a LoadBalancer for the pod")
+				err = iisDeploy.ExposeDeleteIfExist(deploymentPrefix, "default", "LoadBalancer", 80, 80)
+				Expect(err).NotTo(HaveOccurred())
+				iisService, err := service.Get(deploymentName, "default")
+				Expect(err).NotTo(HaveOccurred())
+				err = iisService.WaitForIngress(cfg.LBTimeout, 5*time.Second)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Verifying that the service is reachable and returns the default IIS start page")
+				err = iisService.ValidateWithRetry("(IIS Windows Server)", sleepBetweenRetriesWhenWaitingForPodReady, cfg.Timeout)
+				Expect(err).NotTo(HaveOccurred())
+
+				pods, err := iisDeploy.Pods()
+				Expect(err).NotTo(HaveOccurred())
+				nodeName := pods[0].Spec.NodeName
+				ctx, cancel := context.WithTimeout(context.Background(), 6000*time.Second)
+				defer cancel()
+
+				By("Adding taint to all other Windows nodes")
+				nodeList, err := node.Get()
+				for _, n := range nodeList.Nodes {
+					if n.IsWindows() && n.Metadata.Name != nodeName {
+						n.AddTaint(node.Taint{Key: "key", Value: "value", Effect: "NoSchedule"})
+					}
+				}
+
+				By("Restarting VM " + nodeName + " in resource group " + cfg.ResourceGroup)
+
+				// Getting vmss for the vm
+				vmssPage, err := azureClient.ListVirtualMachineScaleSets(ctx, cfg.ResourceGroup)
+				vmssList := vmssPage.Values()
+
+				// Name of VMSS of nodeName
+				var vmssName string
+				// InstanceID of VM in its VMSS
+				var instanceID string
+				for _, vmss := range vmssList {
+					if !strings.Contains(nodeName, *vmss.Name) {
+						continue
+					}
+					vmName := *vmss.Name + "_" + nodeName[len(nodeName)-1:]
+					vmPage, err := azureClient.ListVirtualMachineScaleSetVMs(ctx, cfg.ResourceGroup, *vmss.Name)
+					Expect(err).NotTo(HaveOccurred())
+
+					vmList := vmPage.Values()
+					for _, vm := range vmList {
+						if vmName == *vm.Name {
+							vmssName = *vmss.Name
+							instanceID = *vm.InstanceID
+							break
+						}
+					}
+					if instanceID != "" {
+						break
+					}
+				}
+
+				instanceIDs := &compute.VirtualMachineScaleSetVMInstanceIDs{&[]string{instanceID}}
+				err = azureClient.RestartVirtualMachineScaleSets(ctx, cfg.ResourceGroup, vmssName, instanceIDs)
+				Expect(err).NotTo(HaveOccurred())
+
+				//Wait for VM to come up
+				time.Sleep(30 * time.Second)
+
+				By("Verifying that the service is still reachable and returns the default IIS start page")
+				err = iisService.ValidateWithRetry("(IIS Windows Server)", sleepBetweenRetriesWhenWaitingForPodReady, cfg.Timeout)
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
If this errors out does would the taints stay around?
This seems like it could negatively impact future test cases.

Reply:
in other languages i'd like to see this get put into a finally block.
Can we add a defer block that would run just around this test case in golang?

Reply:
Yeah I guess if it isn't successful, they won't be removed. I don't see a way around this

Reply:
Deferred in line 2073